### PR TITLE
Bug 2094413: Random suffix for TALO created object names

### PR DESCRIPTION
**Pull request by jc-rh**
Backport for #145

### CODE REVIEW

File: controllers/utils/common.go [hunk 1]
```diff
@@ -26,3 +29,32 @@ func GetMinOf3(number1 int, number2 int, number3 int) int {
 		return number3
 	}
 }
+
+// GetSafeResourceName returns the safename if already allocated in the map and creates a new one if not
+func GetSafeResourceName(name string, clusterGroupUpgrade *ranv1alpha1.ClusterGroupUpgrade, maxLength, spareLength int) string {
```
Comment (sabbir-47) [inline controllers/utils/common.go:34]: is spareLength is always assigned to 0, why would we need it?

Comment (jc-rh) [reply controllers/utils/common.go:34]: it's not always 0

File: controllers/utils/common.go [hunk 2]
```diff
@@ -26,3 +29,32 @@ func GetMinOf3(number1 int, number2 int, number3 int) int {
...
+	return safeName
+}
+
+// NewSafeResourceName creates a safe name to use with random suffix and possible truncation based on limits passed in
+func NewSafeResourceName(name, suffix string, maxLength, spareLength int) string {
+	if suffix == "" {
+		suffix = rand.String(RandomNameSuffixLength)
+	}
+	suffixLength := len(suffix)
+	maxGeneratedNameLength := maxLength - suffixLength - spareLength - 1
```
[the remark this answers is not in the document]
Comment (jc-rh) [reply controllers/utils/common.go:52]: it's the hyphen in front of the random part

[the remark this answers is not in the document]
Comment (sabbir-47) [reply controllers/utils/common.go:52]: Understood.

File: controllers/utils/common.go (hunk 1 above)
Comment (sabbir-47) [reply controllers/utils/common.go:34]: I see  in 2 occassions the function was called:

1. GetSafeResourceName(mcvForInstallPlanName, clusterGroupUpgrade, MaxObjectNameLength, 0)
2. GetSafeResourceName(mcaName, clusterGroupUpgrade, MaxObjectNameLength, 0)

Comment (jc-rh) [reply controllers/utils/common.go:34]: Diff is too big and hidden by default for this file:

```
func (r *ClusterGroupUpgradeReconciler) createNewPolicyFromStructure(
	ctx context.Context, clusterGroupUpgrade *ranv1alpha1.ClusterGroupUpgrade, policy *unstructured.Unstructured) error {

	name := policy.GetName()
	safeName := utils.GetSafeResourceName(name, clusterGroupUpgrade, utils.MaxPolicyNameLength, len(policy.GetNamespace())+1)
```

Comment (sabbir-47) [reply controllers/utils/common.go:34]: Ah! Missed that.

**Review by serngawy — APPROVED** [review]

**Pull request merged**